src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Cache/CacheTag.cs

```diff
@@ -119,13 +117,13 @@ public override async ValueTask<Completion> WriteToAsync(TextWriter writer, Text
             return Completion.Normal;
         }
 
-        private async Task<string> EvaluateStatementsAsync(TextEncoder encoder, TemplateContext context)
+        private static async Task<string> EvaluateStatementsAsync(IReadOnlyList<Statement> statements, TextEncoder encoder, TemplateContext context)
         {
             using (var sb = StringBuilderPool.GetInstance())
             {
                 using (var content = new StringWriter(sb.Builder))
                 {
-                    foreach (var statement in Statements)
+                    foreach (var statement in statements)
```

Comment:
for loop doesn't allocate the boxed enumerator that is needed to enumerate the collection interface, micro optimization